src/metrics-collector/db/sqldb/sqldb_test.go

```diff
@@ -0,0 +1,377 @@
+		BeforeEach(func() {
+			db, err = NewSQLDB(conf, logger)
+			if err != nil {
+				Fail("can not connect database")
+			}
+		})
+
+		AfterEach(func() {
+			dropMetricsTable()
+		})
+
+		Context("When insert a matric", func() {
+			BeforeEach(func() {
+				instances := []metrics.InstanceMetric{{23456312, 0, "3333"}, {23556312, 1, "6666"}}
+				metric := &metrics.Metric{
+					AppId:     "test-app-id",
+					Name:      metrics.MemoryMetricName,
+					Unit:      metrics.UnitBytes,
+					TimeStamp: 11111111,
+					Instances: instances,
+				}
+				err = db.SaveMetric(metric)
+			})
+
+			It("should not error", func() {
+				Expect(err).To(BeNil())
+			})
+
+			It("has the record in database", func() {
+				b := hasMetric("test-app-id", 11111111)
+				Expect(b).To(BeTrue())
+			})
+		})
+
+	})
+
+	Describe("RetrieveMetrics", func() {
+		BeforeEach(func() {
+			db, err = NewSQLDB(conf, logger)
+			if err != nil {
+				Fail("can not connect database")
+			}
+			start = 0
+			end = -1
+			appId = "test-app-id"
+			metricName = metrics.MemoryMetricName
+		})
+
+		AfterEach(func() {
+			dropMetricsTable()
+		})
+
+		Context("When metrics table is empty", func() {
+			BeforeEach(func() {
+				mtrcs, err = db.RetrieveMetrics("test-app-id", metrics.MemoryMetricName, start, end)
+			})
+
+			It("should not error and return empty metrics", func() {
+				Expect(err).To(BeNil())
+				Expect(len(mtrcs)).To(Equal(0))
+			})
+		})
+
+		Context("when metrics table is not empty", func() {
+			JustBeforeEach(func() {
+				instances := []metrics.InstanceMetric{{123456, 0, "3333"}, {123476, 1, "6666"}}
+				metric := &metrics.Metric{
+					AppId:     appId,
+					Name:      metricName,
+					Unit:      metrics.UnitBytes,
+					TimeStamp: 222222,
+					Instances: instances,
+				}
+				db.SaveMetric(metric)
+
+				metric.TimeStamp = 333333
+				db.SaveMetric(metric)
+
+				metric.TimeStamp = 666666
+				db.SaveMetric(metric)
+
+				mtrcs, err = db.RetrieveMetrics("test-app-id", metrics.MemoryMetricName, start, end)
+			})
+
+			Context("when metrics table does not has the metrics from given app", func() {
+				BeforeEach(func() {
+					appId = "other-app-id"
+				})
+
+				It("should not error and return empty metrics", func() {
+					Expect(err).To(BeNil())
+					Expect(len(mtrcs)).To(Equal(0))
+				})
+
+			})
+
+			Context("when metrics table does not has memory metrics", func() {
+				BeforeEach(func() {
+					metricName = "other-metric"
+				})
+
+				It("should not error and return empty metrics", func() {
+					Expect(err).To(BeNil())
+					Expect(len(mtrcs)).To(Equal(0))
+				})
+
+			})
+
+			Context("when end time is -1", func() {
+				BeforeEach(func() {
+					start = 333333
+					end = -1
+				})
+
+				It("should not error and return correct number of metrics", func() {
+					Expect(err).To(BeNil())
+					Expect(len(mtrcs)).To(Equal(2))
+				})
+
+			})
+
+			Context("when start time and endtime are postive numbers", func() {
```

Comment:
remove the context, you haven't set up anything for the tests so it isn't needed.